### PR TITLE
Generate doxygen documentation

### PR DESCRIPTION
**Pull request by outscale-toa**

### CODE REVIEW

File: include/packetgraph/firewall.h [hunk 1]
```diff
@@ -56,12 +54,14 @@ struct pg_brick *pg_firewall_new(const char *name, uint64_t flags,
  *		passed, it will allow all traffic.
  * @param	side side where the filter will occurs, you can use SIDE_MAX
  *		to specify both sides.
+ * @param   stateful allow packet inspection to distinguish legitimate
+ *      packets for different types of connections.
```
Comment (outscale-jju) [inline include/packetgraph/firewall.h]: Indentation seems bad + space at the end

File: include/packetgraph/graph.h [hunk 2]
```diff
@@ -37,11 +37,12 @@ struct pg_graph;
...
  *   parallel.
  *
- * @param	graph graph's name
+ * @param	name graph's name
  * @param	explore if set, explore graph from this brick.
  *		See pg_graph_explore for more information.
  *		Create a empty graph if set to NULL.
- * @return      new allocated and initialized graph, NULL on fail.
+ * @param   error is set in case of an error. 
+ * @return  new allocated and initialized graph, NULL on fail.
```
Comment (outscale-jju) [inline include/packetgraph/graph.h]: Indentation logic ?

File: include/packetgraph/nic.h [hunk 3]
```diff
@@ -25,14 +25,14 @@
  * Create a new nic brick
  * Note: this brick support pg_brick_rx_bytes and pg_brick_tx_bytes.
  *
- * @name:	name of the brick
- * @ifname:	the name of the interface you want to use.
+ * @param	name the brick's name
+ * @param   ifname the name of the interface you want to use.
```
Comment (outscale-jju) [inline include/packetgraph/nic.h]: ident ?

File: include/packetgraph/nic.h [hunk 4]
```diff
@@ -25,14 +25,14 @@
...
  *	This is the same syntax as --vdev in DPDK:
  *	http://pktgen.readthedocs.org/en/latest/usage_eal.html?highlight=vdev
  *	For example, if you want to open a system NIC, put:
  *	"eth_pcap0,iface=eth2"
- * @errp:	set in case of an error
- * @return:	a pointer to a brick structure on success, NULL on error
+ * @param   errp set in case of an error
+ * @return	a pointer to a brick structure on success, NULL on error
```
Comment (outscale-jju) [inline include/packetgraph/nic.h]: ident ?

File: include/packetgraph/nic.h [hunk 5]
```diff
@@ -42,11 +42,10 @@ struct pg_brick *pg_nic_new(const char *name,
...
  *
- * @name:	name of the brick
- * @output:	The side of the output (so the side of the nic)
- * @portid:	the id of the interface you want to use
- * @errp:	set in case of an error
- * @return:	a pointer to a brick structure, on success, 0 on error
+ * @param   name name of the brick
+ * @param   portid the id of the interface you want to use
+ * @param   errp set in case of an error
+ * @return	a pointer to a brick structure, on success, 0 on error
```
Comment (outscale-jju) [inline include/packetgraph/nic.h]: ident ?

File: include/packetgraph/packet.h [hunk 6]
```diff
@@ -25,8 +25,8 @@ IGNORE_NEW_TYPEDEFS typedef struct rte_mbuf pg_packet_t;
 /**
  * Get pointer to packet's data
  *
- * @packet:	packet
- * @return:	a pointer to packet's data.
+ * @param   packet the packet
+ * @return	a pointer to packet's data.
```
Comment (outscale-jju) [inline include/packetgraph/packet.h]: ident ?

File: include/packetgraph/packet.h [hunk 7]
```diff
@@ -36,27 +36,27 @@ uint8_t *pg_packet_data(pg_packet_t *packet);
 /**
  * Get packet's length
  *
- * @packet:	packet
- * @return:	len of the packet in bytes
+ * @param   packet the packet
+ * @return	len of the packet in bytes
```
Comment (outscale-jju) [inline include/packetgraph/packet.h]: ident ?

File: include/packetgraph/packet.h [hunk 8]
```diff
@@ -36,27 +36,27 @@ uint8_t *pg_packet_data(pg_packet_t *packet);
...
 PG_WARN_UNUSED
 unsigned int pg_packet_len(pg_packet_t *packet);
 
 /**
  * Get packet's max length
  *
- * @packet:	packet
- * @return:	maximum len of the packet in bytes
+ * @param   packet the packet
+ * @return	maximum len of the packet in bytes
```
Comment (outscale-jju) [inline include/packetgraph/packet.h]: ident ?

File: include/packetgraph/packet.h [hunk 9]
```diff
@@ -85,7 +86,7 @@ int pg_packet_get_l4_len(pg_packet_t *packet);
  * Compute checksum of an ipv4 header
  *
  * Note: you should set your checksum to 0.
- * @ip_hdr	pointer to ip's header
+ * @param   ip_hdr pointer to ip's header
```
Comment (outscale-jju) [inline include/packetgraph/packet.h]: ident ?

File: include/packetgraph/pmtud.h [hunk 10]
```diff
@@ -24,11 +24,11 @@
...
- * @name:	name of the brick
- * @output:	side where packets can exit without been check
- * @mtu_size:	allowed MTU size (ethernet and protocols above)
- * @errp:	is set in case of an error
- * @return:	a pointer to a brick, NULL on error
+ * @param   name ame of the brick
+ * @param   output side where packets can exit without been check
+ * @param   mtu_size allowed MTU size (ethernet and protocols above)
+ * @param   errp is set in case of an error
+ * @return	a pointer to a brick, NULL on error
```
Comment (outscale-jju) [inline include/packetgraph/pmtud.h]: ident ?

File: include/packetgraph/queue.h [hunk 11]
```diff
@@ -46,12 +46,12 @@
  * Queues have a maximal size, if you try to burst in a full queue, the oldest
  * burst will be free.
  *
- * @name:	name of the brick
- * @size:	maximal size of the queue. If the queue is full, old bursts
+ * @param   name name of the brick
+ * @param   size maximal size of the queue. If the queue is full, old bursts
```
Comment (outscale-jju) [inline include/packetgraph/queue.h]: ident ?

File: include/packetgraph/queue.h [hunk 12]
```diff
@@ -46,12 +46,12 @@
...
  *		will be dropped. If size <= 0, a default queue size of 10 will
  *		be chosen.
- * @error:	is set in case of an error
- * @return:	a pointer to a brick structure on success, NULL on error
+ * @param   error is set in case of an error
+ * @return 	a pointer to a brick structure on success, NULL on error
```
Comment (outscale-jju) [inline include/packetgraph/queue.h]: ident ?

File: include/packetgraph/tap.h [hunk 13]
```diff
@@ -27,12 +27,12 @@
  * If you want a faster interface, you should look at special dpdk interfaces
  * KNI (http://dpdk.org/doc/guides/prog_guide/kernel_nic_interface.html)
  *
- * @name:	name of the brick
- * @ifname:	interface name, set to NULL to automatically get an name.
+ * @param	name of the brick
+ * @param   ifname interface name, set to NULL to automatically get an name.
```
Comment (outscale-jju) [inline include/packetgraph/tap.h]: ident ?

File: include/packetgraph/tap.h [hunk 14]
```diff
@@ -27,12 +27,12 @@
...
 		If provided, interface name must have a max size of IFNAMSIZ.
 		Larger names will be truncated.
- * @errp:	is set in case of an error
- * @return:	a pointer to a brick structure on success, 0 on error
+ * @param   errp is set in case of an error
+ * @return	a pointer to a brick structure on success, 0 on error
```
Comment (outscale-jju) [inline include/packetgraph/tap.h]: ident ?

File: include/packetgraph/tap.h [hunk 15]
```diff
@@ -42,17 +42,18 @@ struct pg_brick *pg_tap_new(const char *name,
 /**
  * Get interface's name.
  *
- * @brick:	brick's pointer
- * @return:	a pointer to interface's name (you MUST NOT free it)
+ * @param   brick brick's pointer
+ * @return	a pointer to interface's name (you MUST NOT free it)
```
Comment (outscale-jju) [inline include/packetgraph/tap.h]: ident ?

File: include/packetgraph/user-dipole.h [hunk 16]
```diff
@@ -43,6 +44,7 @@ typedef void (*pg_user_dipole_callback_t)(struct pg_brick *brick,
  * applications by just providing callbacks to send and receive packets.
  *
  * @param	name name of the brick
+ * @param   callback to send and receive packets
```
Comment (outscale-jju) [inline include/packetgraph/user-dipole.h]: ident ?

File: include/packetgraph/vhost.h [hunk 17]
```diff
@@ -61,10 +61,10 @@ struct pg_brick;
...
  * Note: this brick support pg_brick_rx_bytes and pg_brick_tx_bytes.
  *
- * @name:	name of the brick
- * @flags:      features to pass (see PG_VHOST_USER_*)
- * @errp:	set in case of an error
- * @return:	a pointer to a brick structure on success, NULL on error
+ * @param   name name of the brick
+ * @param   flags features to pass (see PG_VHOST_USER_*)
+ * @param   errp set in case of an error
+ * @return	a pointer to a brick structure on success, NULL on error
```
Comment (outscale-jju) [inline include/packetgraph/vhost.h]: ident ?

File: include/packetgraph/vtep.h [hunk 18]
```diff
@@ -53,6 +53,7 @@ struct ether_addr *pg_vtep_get_mac(struct pg_brick *brick);
  * @param	vni the VNI to add, must be < 2^24
  * @param	multicast_ip the multicast ip to associate to the VNI
  * @param	errp an error pointer
+ * @return  0 on success, -1 on error
```
Comment (outscale-jju) [inline include/packetgraph/vtep.h]: ident ?

File: include/packetgraph/vtep.h [hunk 19]
```diff
@@ -67,6 +68,7 @@ int pg_vtep4_add_vni(struct pg_brick *brick,
...
```
Comment (outscale-jju) [inline include/packetgraph/vtep.h]: ident ?

File: README.md [hunk 20]
```diff
@@ -2,7 +2,11 @@
 
 Packetgraph is a library aiming to give the user a tool to build networks graph easily, It's built upon the fast [DPDK library](http://dpdk.org/).
 
-[![Build Status](https://travis-ci.org/outscale/packetgraph.svg?branch=master)](https://travis-ci.org/outscale/packetgraph)
+\htmlonly
+<a href="https://travis-ci.org/outscale/packetgraph">
+<img src="https://travis-ci.org/outscale/packetgraph.svg">
+</a>
+\endhtmlonly
```
Comment (outscale-jju) [inline README.md]: I really don't like this solution.
Would it be possible to run a markdown -> html converter and pass the result to doxygen ?

File: .travis.sh [hunk 21]
```diff
@@ -22,6 +22,7 @@ make tests-pmtud
 make tests-tap
 make tests-ip-fragment
 make tests-thread
+make doc
```
Comment (outscale-jju) [inline .travis.sh]: Does travis will fail in case of Warning or Error during doxygen generation ? Did you test it ?

File: src/firewall.c [hunk 22]
```diff
@@ -98,14 +98,14 @@ static inline int firewall_side_to_npf_rule(enum pg_side side)
 }
 
 int pg_firewall_rule_add(struct pg_brick *brick, const char *filter,
-			 enum pg_side dir, int stateful, struct pg_error **errp)
+			 enum pg_side side, int stateful, struct pg_error **errp)
```
Comment (outscale-jju) [inline src/firewall.c]: WARNING:LONG_LINE: line over 80 characters

Did you run tests ?

File: .travis.sh (hunk 21 above)
Comment (outscale-toa) [inline .travis.sh]: Yes, now travis will fail in case of Warning or Error during doxygen generation.

**Pull request merged**